legate/cpp/tile_io.cpp

```diff
@@ -105,6 +106,71 @@ struct tile_read_write_fn {
   }
 };
 
+/**
+ * @brief Flatten the domain point to a 1D point
+ *
+ * @param lo_dp Lower point
+ * @param hi_dp High point
+ * @param point_dp The domain point to flatten
+ * @return The flatten domain point
+ */
+template <int32_t DIM>
+size_t linearize(const legate::DomainPoint& lo_dp,
+                 const legate::DomainPoint& hi_dp,
+                 const legate::DomainPoint& point_dp)
+{
+  legate::Point<DIM> lo      = lo_dp;
+  legate::Point<DIM> hi      = hi_dp;
+  legate::Point<DIM> point   = point_dp;
+  legate::Point<DIM> extents = hi - lo + legate::Point<DIM>::ONES();
+  size_t idx                 = 0;
+  for (int32_t dim = 0; dim < DIM; ++dim) {
+    idx = idx * extents[dim] + point[dim] - lo[dim];
+  }
```

Comment:
Done. 
`point_dp` must be within `lo` and `hi`. Since they come from `context.get_task_index()` and `context.get_launch_domain()` directly , I don't we need a check here.